### PR TITLE
make tests for empty inputs check zero parameter grads

### PR DESCRIPTION
**Pull request by ngimel**
Make batch norm with empty inputs return zero parameter gradients. Now batch norm, group norm and convolutions now return zero grads for parameters, so make tests check that. Fixes some bullet points in #12013 (interpolate is not fixed by this PR, is being fixed in other PRs)

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 5a0a1b84:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=821b6aa769645c8190703b7d8e2cc9f36597853a&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base 821b6aa7 since Jan 29
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=5a0a1b84e93d055ef0ed4a160f5419a73c6372a6).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_linux_backward_compatibility_check_test](https://dr.pytorch.org/build-details.html?build_id=93079155) from Jan 29



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: aten/src/ATen/native/Normalization.cpp [hunk 1]
```diff
@@ -532,7 +532,11 @@ Tensor batch_norm(
     const Tensor& running_mean /* optional */, const Tensor& running_var /* optional */,
     bool training, double momentum, double eps, bool cudnn_enabled) {
   if (input.numel()==0){
-    return input; //return input instead of new empty tensor, because new empty tensor breaks the gradient chain
+    //don't return view of input, don't return empty tensor because it will break gradient chain
+    auto out = input.clone();
+    if (weight.defined()) out = out*weight[0];
```
Comment (albanD) [inline aten/src/ATen/native/Normalization.cpp]: nit: space to be consistent with line below

File: test/test_nn.py [hunk 2]
```diff
@@ -8983,18 +8983,17 @@ def _test_GroupNorm_cuda_half(self):
         output.sum().backward()
         self.assertEqual(output.type(), input.type())
 
-    def _test_module_empty_input(self, module, inp, check_size=True):
+    def _test_module_empty_input(self, module, inp, check_size=True, check_parameter_grads=True):
```
Comment (albanD) [inline test/test_nn.py]: I guess `check_parameter_grads` is an artefact of an older version? It does not seem to be used.

**Review by albanD — CHANGES_REQUESTED** [review]
I think you're missing the test for batchnorm no?
Otherwise looks good.

File: test/test_nn.py (hunk 2 above)
Comment (ngimel) [reply test/test_nn.py]: It is not used, but it might be if someone adds support for new ops without taking care of gradients. I'll remove it for now though.

**Comment by ngimel** [comment]
> I think you're missing the test for batchnorm no?

Tests for batch norm were already present, they just did not check parameter gradients.

**Review by albanD — APPROVED** [review]
Right I was confused because you add a test for convtranspose :)
